Source/WebKit/Platform/spi/visionos/WKSurroundingsEffect.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2025 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+#if os(visionOS)
+
+import Combine
+@_weakLinked internal import SwiftUI
+import UIKit
+
+@objc(WKSurroundingsEffectType)
+@available(visionOS 1.0, *)
+enum WKSurroundingsEffectType: Int {
+    case none = 0
+    case semiDark = 1
+    case dark = 2
+    case ultraDark = 3
+}
+
+@MainActor
+@objc(WKSurroundingsEffectManager)
+@available(visionOS 1.0, *)
+class WKSurroundingsEffectManager: NSObject, ObservableObject {
+    @objc
+    static let shared = WKSurroundingsEffectManager()
+
+    @Published
+    @objc
+    dynamic var currentEffect: WKSurroundingsEffectType = .none
```

Comment:
@Published and `dynamic` can be removed once Observation is used instead of Combine